File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(s) {
    let stack=[];

    for(let i=0;i<s.length;i++){
        if(s[i]=="+" || s[i]=="-" || s[i]=="*"||s[i]=="/"){
           let a=stack.pop();
           let b=stack.pop();
           let ans=eval(`${b} ${s[i]} ${a}`);
           stack.push(Math.trunc(ans));
        }
        else {
            stack.push(s[i]);
        }
    }
    return Number(stack.pop());
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] arr) {

        Stack<Integer> stack=new Stack<>();
        Map<String,BiFunction<Integer,Integer,Integer>> map=new HashMap<>();
        map.put("+",(a,b)->b+a);
        map.put("-",(a,b)->b-a);
        map.put("*",(a,b)->b*a);
        map.put("/",(a,b)->b/a);
        for(int i=0;i<arr.length;i++){
            if(map.containsKey(arr[i])){
                int a=stack.pop();
                int b=stack.pop();
                int ans=map.get(arr[i]).apply(a,b);
                stack.push(ans);
            }else{
                stack.push(Integer.parseInt(arr[i]));
            }
        }
        return stack.pop();
        
    }
}